examples/gno.land/p/demo/grc/grc721/grc721_royalty.gno

```diff
@@ -0,0 +1,78 @@
+package grc721
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+// royaltyNFT represents a non-fungible token (NFT) with royalty functionality.
+type royaltyNFT struct {
+	*metadataNFT          // Embedding metadataNFT for NFT functionality
+	tokenRoyaltyInfo avl.Tree  // AVL tree to store royalty information for each token
+}
+
+// Ensure that royaltyNFT implements the IGRC2981 interface.
+var _ IGRC2981 = (*royaltyNFT)(nil)
+
+// NewNFTWithRoyalty creates a new royalty NFT with the specified name, symbol, and royalty calculator.
+func NewNFTWithRoyalty(name string, symbol string) *royaltyNFT {
+	// Create a new NFT with metadata
+	nft := NewNFTWithMetadata(name, symbol)
+
+	return &royaltyNFT{
+		metadataNFT:        nft,
+		tokenRoyaltyInfo:   avl.Tree{},
+	}
+}
+
+// SetTokenRoyalty sets the royalty information for a specific token ID.
+func (r *royaltyNFT) SetTokenRoyalty(tid TokenID, royaltyInfo RoyaltyInfo) (bool, error) {
+	// Validate the payment address
+	if err := isValidAddress(royaltyInfo.PaymentAddress); err != nil {
+		return false, ErrInvalidRoyaltyPaymentAddress
+	}
+
+	// Check if royalty percentage exceeds 100%
+	if royaltyInfo.Percentage > 100 {
```

Comment:
Nit: extract `100` to a constant, say `maxRoyaltyPercentage`

Reply:
https://github.com/linhpn99/gno/commit/aebba7474ad24f3d44ceced64b7fe4a02fc85084